eth/db/journal.py

```diff
@@ -211,11 +221,14 @@ def __setitem__(self, key: bytes, value: bytes) -> None:
 
     def _exists(self, key: bytes) -> bool:
         val = self.get(key)
-        return val is not None and val is not DELETED_ENTRY
+        return val is not None and val not in (ERASE_CREATED_ENTRY, DELETED_ENTRY)
 
     def __delitem__(self, key: bytes) -> None:
         self.latest[key] = DELETED_ENTRY
 
+    def local_delete(self, key: bytes) -> None:
+        self.latest[key] = ERASE_CREATED_ENTRY
+
```

Comment:
nitpick: It's a little weird and unintuitive that `DELETED_ENTRY` and `ERASE_CREATED_ENTRY` have such different APIs. Something like dropping `__delitem__` and adding `delete(key: bytes, exists_in_db: bool = False)` might be easier for future readers to understand.